modules/ROOT/pages/provisioning-hetzner.adoc

```diff
@@ -0,0 +1,173 @@
+= Provisioning Fedora CoreOS on Hetzner
+
+This guide shows how to provision new Fedora CoreOS (FCOS) nodes on Hetzner.
+Fedora CoreOS is currently not available as an option in the operating system selection on Hetzner.
+Fedora CoreOS images for Hetzner are currently not directly available for download either.
+Thus you must first download a Fedora CoreOS RAW disk image, then convert it to an Hetzner image locally and finally create a snapshot from it in your Hetzner account using the https://github.com/apricote/hcloud-upload-image[hcloud-upload-image] tool.
```

Comment:
Two official alternatives:

- [`hcloud` CLI](https://github.com/hetznercloud/cli) and some shell script
- Packer with [`packer-plugin-hcloud`](https://github.com/hetznercloud/packer-plugin-hcloud). Example templates:
  - [Flatcar](https://www.flatcar.org/docs/latest/installing/cloud/hetzner/#template)
  - [Talos](https://www.talos.dev/v1.7/talos-guides/install/cloud-platforms/hetzner/#packer)

I personally found both of these options annoying and not very user friendly, so I wrote the CLI/Go Lib to improve on this. `hcloud-upload-image` will be deprecated in favor of an `hcloud` subcommand if an official endpoint for uploading images is added to the API.

Reply:
I would say this entire page is "best-effort" given that we don't produce images. But I'm also fine adding a warning specifically for this tool.

Reply:
I did not go the packer route because it depends on packer and it's no longer open source.

Reply:
It should also be mentioned that all variants ultimately create a temporary server that will be billed at the end of the month (even though it shouldn't be that much money since Hetzner charges by minutes used; however, unknown servers unexpectedly appearing on the bill might confuse users). It should also be mentioned that this will create a snapshot that also costs money depending on the image size (0.0110 € per GB per month plus VAT); permanently if not deleted manually.

Reply:
I added an "important" note about both of those topics at the top.